cmd/snap-update-ns/utils.go

```diff
@@ -58,53 +138,29 @@ var (
 // after each segment is created and opened. The special value -1 may be used
 // to request that ownership is not changed.
 func secureMkdirAll(name string, perm os.FileMode, uid, gid int) error {
-	// Declare var and don't assign-declare below to ensure we don't swallow
-	// any errors by mistake.
-	var err error
-	var fd int
-
-	const openFlags = syscall.O_NOFOLLOW | syscall.O_CLOEXEC | syscall.O_DIRECTORY
+	logger.Debugf("secure-mkdir-all %q %v %d %d", name, perm, uid, gid)
 
 	// Only support absolute paths to avoid bugs in snap-confine when
 	// called from anywhere.
 	if !filepath.IsAbs(name) {
 		return fmt.Errorf("cannot create directory with relative path: %q", name)
 	}
-	// Open the root directory and start there.
-	fd, err = sysOpen("/", openFlags, 0)
+	segments := strings.FieldsFunc(filepath.Clean(name), func(c rune) bool { return c == '/' })
+
```

Comment:
You should probably check if `len(segments) > 0`. Haven't tracked through the code but I think you'll get a panic with `name = "/"`.